File: lib/features/webprint/encripter.dart
import 'dart:convert';
import 'dart:typed_data';

import 'package:basic_utils/basic_utils.dart';
import 'package:pointycastle/export.dart';

/// 名城大 Web プリントの公開鍵 (PKCS#1 v1.5 / RSA) で平文を暗号化し
/// Base64 エンコードして返す。
///
/// ブラウザ側 (JSEncrypt 互換) と一致させるため OAEP ではなく v1.5 を使う。
/// 入力は単一ブロックに収まる前提 (ユーザ名/パスワード相当)。
String encryptForMeijo(String message, String publicKeyPem) {
  final publicKey = CryptoUtils.rsaPublicKeyFromPem(publicKeyPem);
  final cipher = PKCS1Encoding(RSAEngine())
    ..init(true, PublicKeyParameter<RSAPublicKey>(publicKey));
  final input = Uint8List.fromList(utf8.encode(message));
  if (input.length > cipher.inputBlockSize) {
    throw ArgumentError(
      '平文が公開鍵のブロックサイズを超えています '
      '(input=${input.length}, blockSize=${cipher.inputBlockSize})',
    );
  }
  final encrypted = cipher.process(input);
  return base64.encode(encrypted);
}

File: features/webprint/encripter.py
import base64
from Crypto.PublicKey import RSA
from Crypto.Cipher import PKCS1_v1_5 # OAEPから変更

def encrypt_for_meijo(message, public_key_pem):
   # 1. 公開鍵を読み込む
   recipient_key = RSA.import_key(public_key_pem)
   
   # 2. ブラウザと同じ PKCS#1 v1.5 パディングを使用する
   cipher_rsa = PKCS1_v1_5.new(recipient_key)
   
   # 3. 暗号化を実行
   encrypted_data = cipher_rsa.encrypt(message.encode('utf-8'))
   
   # 4. Base64でエンコード
   return base64.b64encode(encrypted_data).decode('utf-8')